fe2-android/docs/README.md

```diff
@@ -179,44 +170,40 @@ The communication stack within the PoP project is made
-  the [Scarlet](https://github.com/Tinder/Scarlet) library.
-* The `LAORepository` exposes the [JSON-RPC](https://www.jsonrpc.org/specification) -based API that
-  provides the publish/subscribe communication abstraction. In it, you will find functions to
-  publish a message, subscribe and unsubscribe to a channel, and so on.
+The [remote module](https://github.com/dedis/popstellar/tree/master/fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote)
+contains most of the network communication logic, it is organized as follows:\
+The `GlobalNetworkManager` is responsible for holding, creating and destroying WebSocket connections.\
+It uses the `ConnectionFactory` to produce new `Connection` objects using
+the [Scarlet](https://github.com/Tinder/Scarlet) library.
 
-The application-level message generation logic is implemented in the `ViewModel` classes, for
-example the request for a LAO creation is in `HomeViewModel`. In the future an API could be created
-to simplify the logic in the application code and reduce repetition.
+The `Connection` is passed to a `LAOConnectionManager` that is responsible for handling the JsonRPC layer of the
+protocol. It exposes the protocol API with the `MessageSender` interface that can be retrieved by `ViewModel`though
+the `GlobalNetworkManager`.
 
 For more information on sending messages on the network, please refer to
-the [repository module](https://github.com/dedis/popstellar/tree/master/fe2-android/app/src/main/java/com/github/dedis/popstellar/repository)
+the [remote module](https://github.com/dedis/popstellar/tree/master/fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote)
 and the `ViewModel` classes. Also, make sure you have a solid understanding
 of [JSON-RPC](https://www.jsonrpc.org/specification),
 the [Protocol Specifications](https://docs.google.com/document/d/1fyNWSPzLhM6W9V0VTFf2waMLiJGcscy7wa4bQlLkySM)
-and their actual implementation in
-the [protocol schemas](https://github.com/dedis/popstellar/tree/master/protocol)
+and their actual implementation in the [protocol schemas](https://github.com/dedis/popstellar/tree/master/protocol)
 
 ### Getting messages over the wire
 
 Once it is clear how to send messages over the wire, it is important to turn one's attention to
 receiving them.
 
-Because receiving messages over the network and processing them to update the application state are
-very different steps involving completely unrelated logic.
+Because receiving messages over the network and processing them to update the application state are very different steps
+involving completely unrelated logic.
 
-On the networking side, the `repository/remote` module observes the upcoming messages and notifies
-the `LAORepository`.
+On the networking side, the embedded Scarlet service handles the upcoming websocket messages and transform them into
+Java Object using Gson as the Json Parser.
```

Comment:
I like how you've structured the coming lines as "following a message" through the network stack. I'd make it even more explicit.

```suggestion
When the network dispatches a WebSocket message to the application, the embedded Scarlet service handles it and transforms it into
a Java Object using Gson as the Json Parser.
```